src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1070,28 +1084,34 @@ internal static MethodSymbol TryFindBaseImplementationInInterface(NamedTypeSymbo
         private static Symbol FindMostSpecificImplementation(Symbol interfaceMember, NamedTypeSymbol implementingInterface)
         {
             HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-            var implementingMember = FindImplementationInInterface(interfaceMember, implementingInterface) ??
-                                     FindMostSpecificImplementation(interfaceMember, implementingInterface,
-                                                                    ref useSiteDiagnostics,
-                                                                    out var _, out var _);
-            return implementingMember;
+            MultiDictionary<Symbol, Symbol>.ValueSet implementingMember = FindImplementationInInterface(interfaceMember, implementingInterface);
+
+            switch (implementingMember.Count)
+            {
+                case 0:
+                    return FindMostSpecificImplementationInBases(interfaceMember, implementingInterface,
+                                                                 ref useSiteDiagnostics,
+                                                                 out var _, out var _);
+                case 1:
+                    return implementingMember.Single();
+                default:
+                    return null;
+            }
         }
 
         /// <summary>
         /// One implementation M1 is considered more specific than another implementation M2 
         /// if M1 is declared on interface T1, M2 is declared on interface T2, and 
         /// T1 contains T2 among its direct or indirect interfaces.
         /// </summary>
-        private static Symbol FindMostSpecificImplementation(Symbol interfaceMember,
-                                                             TypeSymbol implementingType,
-                                                             ref HashSet<DiagnosticInfo> useSiteDiagnostics,
-                                                             out Symbol conflictingImplementation1,
-                                                             out Symbol conflictingImplementation2)
+        private static Symbol FindMostSpecificImplementationInBases(
+            Symbol interfaceMember,
+            TypeSymbol implementingType,
+            ref HashSet<DiagnosticInfo> useSiteDiagnostics,
+            out Symbol conflictingImplementation1,
+            out Symbol conflictingImplementation2)
         {
-            conflictingImplementation1 = null;
-            conflictingImplementation2 = null;
-            Symbol implementation = null;
-            PooledHashSet<NamedTypeSymbol> shadowedInterfaces = null;
+            var implementations = ArrayBuilder<(MultiDictionary<Symbol, Symbol>.ValueSet MethodSet, MultiDictionary<NamedTypeSymbol, NamedTypeSymbol> Bases)>.GetInstance();
```

Comment:
We should send this to Mads as a reason why we need to have first class type aliases in the language.